core/src/test/java/org/bitcoinj/base/BitcoinNetworkTest.java

```diff
@@ -79,4 +84,47 @@ public void fromIdString_uppercase() {
     public void fromIdString_notExisting() {
         assertFalse(BitcoinNetwork.fromIdString("a.b.c").isPresent());
     }
+
+    @Test
+    public void testLegacyAddressValidity() {
+        LegacyAddress m = LegacyAddress.fromBase58("17kzeh4N8g49GFvdDzSf8PjaPfyoD1MndL", MAINNET);
+        LegacyAddress t = LegacyAddress.fromBase58("n4eA2nbYqErp7H6jebchxAN59DmNpksexv", TESTNET);
+        assertTrue(MAINNET.isValidAddress(m) &&
+                TESTNET.isValidAddress(t) &&
+                SIGNET.isValidAddress(t) &&
+                REGTEST.isValidAddress(t));
+        assertFalse(MAINNET.isValidAddress(t) &&
+                TESTNET.isValidAddress(m) &&
+                SIGNET.isValidAddress(m) &&
+                REGTEST.isValidAddress(m));
```

Comment:
I assume these are meant to use `||` rather than `&&`? But again, I'd prefer single statements.

Reply:
I think the idea here is to assert true only if all are true. Hence, && is appropriate.

Reply:
single assert will indeed be easier to read. (Although lots of assert statements : D )

Reply:
I'll make them single asserts

Reply:
Fixed.